### PR TITLE
Use SecurityTransforms for RSA+PSS

### LINKED ISSUE #82714: RSA-PSS on iOS should use native APIs
Our iOS / tvOS implementation of RSA-PSS currently uses raw RSA, which we pad ourselves. This was due to the RSA-PSS support missing on iOS 10.

https://github.com/dotnet/runtime/blob/7414af2a5f6d8d99efc27d3f5ef7a394e0b23c42/src/native/libs/System.Security.Cryptography.Native.Apple/pal_signverify.c#L126-L127

However, as of https://github.com/dotnet/runtime/pull/81965, the minimum iOS / tvOS version is 11. We should enable the native code paths to do RSA+PSS where possible.

### PR DESCRIPTION
**Pull request by vcsjones**
In #81965 some native code for RSA+PSS was uncommented since the iOS version minimum got bumped to 11. However, we were never actually sending `PAL_SignatureAlgorithm.RsaPss` to the native shim, so it was effectively dead code.

This change allows SecurityTransforms to perform RSA+PSS operations. However, Apple's implementation does not support PSS+MD5. There isn't even an "legacy" identifier to use like we have with PKCSv1.5. In that situation, we will continue to use raw RSA and perform the padding ourselves. Still though, our ethos is to always use the native platform for cryptography when possible, and SHA1+SHA2 for RSA+PSS is the 99% case.

As a bonus, this improves allocations for the non-MD5 algorithms.

|     Method |        Job | Toolchain | SignaturePadding | HashAlgorithmName |        Mean |    Error |   StdDev | Ratio |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------- |----------- |---------- |----------------- |------------------ |------------:|---------:|---------:|------:|-------:|------:|------:|----------:|
|   SignHash | Job-GVNWDB |    branch |              Pss |               MD5 | 1,152.28 us | 1.066 us | 0.945 us |  1.00 |      - |     - |     - |     217 B |
|   SignHash | Job-OVCQWP |      main |              Pss |               MD5 | 1,151.95 us | 0.904 us | 0.755 us |  1.00 |      - |     - |     - |     217 B |
|            |            |           |                  |                   |             |          |          |       |        |       |       |           |
| VerifyHash | Job-GVNWDB |    branch |              Pss |               MD5 |    31.40 us | 0.074 us | 0.065 us |  1.00 |      - |     - |     - |     216 B |
| VerifyHash | Job-OVCQWP |      main |              Pss |               MD5 |    31.41 us | 0.067 us | 0.063 us |  1.00 |      - |     - |     - |     216 B |
|            |            |           |                  |                   |             |          |          |       |        |       |       |           |
|   SignHash | Job-GVNWDB |    branch |              Pss |            SHA256 | 1,152.75 us | 4.020 us | 3.760 us |  1.01 |      - |     - |     - |      65 B |
|   SignHash | Job-OVCQWP |      main |              Pss |            SHA256 | 1,146.19 us | 1.119 us | 0.992 us |  1.00 |      - |     - |     - |     217 B |
|            |            |           |                  |                   |             |          |          |       |        |       |       |           |
| VerifyHash | Job-GVNWDB |    branch |              Pss |            SHA256 |    24.83 us | 0.020 us | 0.017 us |  0.97 |      - |     - |     - |      32 B |
| VerifyHash | Job-OVCQWP |      main |              Pss |            SHA256 |    25.72 us | 0.089 us | 0.069 us |  1.00 | 0.0305 |     - |     - |     216 B |

Closes #82714

### CODE REVIEW

**Review by filipnavara — APPROVED** [review]

**Comment by stephentoub** [comment]
I assume we have tests covering all of this well?

**Comment by vcsjones** [comment]
> I assume we have tests covering all of this well?

Yes. For example, starting here:

https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/SignVerify.cs#L886

and many of the other tests in that file, and others. Higher abstractions like X509 test this additionally.

File: src/libraries/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs [hunk 1]
```diff
@@ -496,19 +492,29 @@ public override bool VerifyHash(ReadOnlySpan<byte> hash, ReadOnlySpan<byte> sign
...
                     return Interop.AppleCrypto.VerifySignature(
                         GetKeys().PublicKey,
                         hash,
                         signature,
                         palAlgId,
-                        Interop.AppleCrypto.PAL_SignatureAlgorithm.RsaPkcs1);
+                        signatureAlgorithm);
                 }
-                else if (padding.Mode == RSASignaturePaddingMode.Pss)
+                else
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs:515]: Nit: remove the else and unindent the block's contents?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs]: Yeah. That looks nicer.

**Review by akoeplinger — APPROVED** [review]

**Review by jozkee — APPROVED** [review]

**Pull request merged**